.github/workflows/docs.yaml

```diff
@@ -0,0 +1,134 @@
+name: Docs
+
+on:
+  push:
+    branches:
+    - 'master'
+    - 'v[0-9]+.[0-9]+'
+    paths:
+    - 'docs/**'
+    - '.github/workflows/docs.yaml'
+  pull_request:
+    branches:
+    - '**'
+    paths:
+    - 'docs/**'
+    - '.github/workflows/docs.yaml'
+    types:
+    - opened
+    - edited
+    - reopened
+    - synchronize
+  schedule:
+  - cron:  '0 23 * * *' # daily at 11pm
+
+env:
+  retention_days: 90
+
+jobs:
+  build-and-test:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        persist-credentials: false
+        fetch-depth: 0
+
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+
+    - name: Set up Poetry
+      run: |
+        set -euExo pipefail
+        shopt -s inherit_errexit
+        
+        ./docs/hack/install-poetry.sh
+
+    - name: Install dependencies
+      run: |
+        set -euExo pipefail
+        shopt -s inherit_errexit
+        
+        pushd ./docs/
+        ~/.poetry/bin/poetry install
+
+    - name: Build docs
+      run: |
+        set -euExo pipefail
+        shopt -s inherit_errexit
+        
+        make -C ./docs/ multiversion SPHINXOPTS="--keep-going"
```

Comment:
multiversion builds from multiple branches specified in config. Changes in different branch shouldn't affect version we are testing here. And it doesn't build actual PR changes for PR builds.